### PR TITLE
Update Method Interventions & Add New Method.

### PR DESCRIPTION
**Pull request by menriquez-IDM**
### Type(s) of change
- [ ] Bugfix
- [ ] Refactor
- [x] New feature
- [ ] Other

### Checklist
- My code follows the [style guide](https://github.com/amath-idm/styleguide)
  - [ ] Yes
  - [ ] N/A
- I've commented my code
  - [x] Yes
  - [ ] N/A
- I've incremented the version number
  - [x] Yes
  - [ ] N/A
- I've updated the changelog
  - [x] Yes
  - [ ] N/A
- I've added tests and checked code coverage
  - [x] Yes
  - [ ] N/A

### 1. MethodIntervention Wrapper API (`fpsim/wrappers.py`)
A new high-level interface for modeling contraceptive interventions:
- **Efficacy changes**: Model improved method effectiveness
- **Duration changes**: Model continuation improvements (counseling, side effect management)
- **Method mix**: Model shifts in method choice (social marketing campaigns)
- **Probability of use**: Model changes in overall contraceptive uptake
- **Switching patterns**: Model transitions between methods

### 2. Dynamic Method Addition
Add new contraceptive methods mid-simulation as interventions:
- Automatic switching matrix expansion
- Flexible configuration of switching patterns
- Dynamic array resizing for results tracking
- Integration with all contraception modules (StandardChoice, SimpleChoice, RandomChoice)

### 3. Comprehensive Testing
- `test_methodintervention.py`: 
- `test_add_new_method.py`: 
- Full coverage of initialization, validation, and simulation integration

### CODE REVIEW

File: fpsim/wrappers.py [hunk 1]
```diff
@@ -0,0 +1,1216 @@
...
+__all__ = [
+    'MethodIntervention',
+    'make_update_methods',
+    'MethodName',
+]
+
+# ==========================================
+# Enums and Type Definitions
+# ==========================================
+class MethodName(str, Enum):
```
Comment (robynstuart) [inline fpsim/wrappers.py]: Should probabaly avoid defining these in >1 place. They're currently defined in `make_method_list()` within methods.py.

File: fpsim/wrappers.py [hunk 2]
```diff
@@ -0,0 +1,1216 @@
...
+    def values(cls) -> List[str]:
+        """Return list of all method name values."""
+        return [m.value for m in cls]
+    
+    @classmethod
+    def excluding_none(cls) -> List[str]:
+        """Return list of method names excluding 'none'."""
+        return [m.value for m in cls if m != cls.NONE]
+
+class PreviewDict(TypedDict, total=False):
```
Comment (robynstuart) [inline fpsim/wrappers.py]: I don't think I understand why we need this... is it for mouseover purposes?

File: fpsim/wrappers.py [hunk 3]
```diff
@@ -0,0 +1,1216 @@
...
+    efficacy: Optional[Dict[str, float]]
+    duration_months: Optional[Dict[str, Union[int, float]]]
+    p_use: Optional[float]
+    method_mix: Optional[Dict[str, float]]
+    switching_matrix: Optional[str]
+    new_method: Optional[Dict[str, Any]]
+    label: Optional[str]
+
+@dataclass(slots=True)
+class InterventionConfig:
```
Comment (robynstuart) [inline fpsim/wrappers.py]: Seems like it'd be safe to use dictionaries of inputs in most cases - what's the additional benefit of this class?

File: fpsim/wrappers.py [hunk 4]
```diff
@@ -0,0 +1,1216 @@
...
+    
+    path = Path(source) if isinstance(source, str) else source
+    
+    if path.suffix != '.json':
+        raise ValueError(f'Unsupported file type: {path.suffix}. Expected .json')
+    
+    return json.loads(path.read_text())
+
+@contextmanager
+def _print_if(condition: bool, message: str):
```
Comment (robynstuart) [inline fpsim/wrappers.py]: Minor comment, but seems like this should live in Starsim (if generally useful) or fp.utils

File: fpsim/wrappers.py [hunk 5]
```diff
@@ -0,0 +1,1216 @@
...
+        - Better follow-up and support services
+        - Community education reducing stigma
+        
+        Parameters
+        ----------
+        method : str
+            Method name: 'pill', 'iud', 'inj', 'impl', 'cond', 'btl', 'wdraw', etc.
+        months : int or float
+            Average duration in months that people stay on the method
+            Examples: 12 = 1 year, 24 = 2 years, 36 = 3 years
```
Comment (robynstuart) [inline fpsim/wrappers.py]: Ideally these should all use Starsim's time logic, so it'd be something like ss.months(30) and then it'd get translated to the right number of timesteps depending on what `dt` is.

File: tests/test_methodintervention.py [hunk 6]
```diff
@@ -0,0 +1,1180 @@
...
+    
+    assert isinstance(intv, fp.update_methods)
+    assert hasattr(intv.pars, 'method_mix')
+
+
+# =============================================================================
+# CHALLENGING EDGE CASE TESTS
+# =============================================================================
+
+def test_overwrite_efficacy_same_method():
```
Comment (robynstuart) [inline tests/test_methodintervention.py]: Do we need all of these? And is there any reason why this wouldn't work?

File: tests/test_methodintervention.py [hunk 7]
```diff
@@ -0,0 +1,1180 @@
...
+    
+    intv = mod.build()
+    assert intv.pars.dur_use['BTL'] == 600.0
+
+
+def test_fractional_duration():
+    """Test fractional month durations."""
+    mod = fp.MethodIntervention(year=2025)
+    
+    mod.set_duration_months('cond', 0.5)  # Half a month
```
Comment (robynstuart) [inline tests/test_methodintervention.py]: Should be able to do things like `mod.set_duration_months('cond', ss.days(1))` as well

File: tests/test_methodintervention.py [hunk 8]
```diff
@@ -0,0 +1,1180 @@
...
+    
+    # Verify both changes were applied to intervention
+    assert intv.pars.eff['Pill'] == 0.98
+    assert intv.pars.dur_use['Pill'] == 30.0
+    
+    # Run simulation to ensure it completes
+    sim = fp.Sim(pars=base_pars, interventions=intv, label='Intervention')
+    sim.run()
+    
+    assert sim.complete
```
Comment (robynstuart) [inline tests/test_methodintervention.py]: Again, we need to check what changed within the sim don't we?

File: tests/test_add_new_method.py [hunk 9]
```diff
@@ -0,0 +1,879 @@
...
+    new_method = fp.Method(
+        name='new_inj',
+        label='New Injectable',
+        efficacy=0.98,
+        modern=True,
+        dur_use=fp.methods.ln(8, 2)
+    )
+    
+    mod = fp.MethodIntervention(year=2010)
+    mod.add_method(
```
Comment (robynstuart) [inline tests/test_add_new_method.py]: I'm curious whether this also supports a workflow where people want to manually upload their own switching matrix
